collect_app/src/main/java/org/odk/collect/android/notifications/NotificationManagerNotifier.java

```diff
@@ -35,18 +42,26 @@
 
     private final Application application;
     private final NotificationManager notificationManager;
+    private final PreferencesProvider preferencesProvider;
 
     private static final int FORM_UPDATE_NOTIFICATION_ID = 0;
     private static final int FORM_SYNC_NOTIFICATION_ID = 1;
     private static final int AUTO_SEND_RESULT_NOTIFICATION_ID = 1328974928;
 
-    public NotificationManagerNotifier(Application application) {
+    public NotificationManagerNotifier(Application application, PreferencesProvider preferencesProvider) {
         this.application = application;
         notificationManager = (NotificationManager) application.getSystemService(NOTIFICATION_SERVICE);
+        this.preferencesProvider = preferencesProvider;
     }
 
     @Override
-    public void onUpdatesAvailable() {
+    public void onUpdatesAvailable(List<ServerFormDetails> updates) {
+        SharedPreferences metaPrefs = preferencesProvider.getMetaSharedPreferences();
```

Comment:
This is the key change here: switch away from using the database to record the versions the user is notified about and also making it more of a UI's (the `Notifier`) concern.